src/libraries/System.Net.Quic/tests/FunctionalTests/QuicListenerTests.cs

```diff
@@ -47,7 +47,10 @@ public async Task Listener_IPv6Any_Accepts_IPv4()
         {
             await Task.Run(async () =>
             {
-                using QuicListener listener = CreateQuicListener(new IPEndPoint(IPAddress.IPv6Any, 0));
+                // Don't use the static IPAddress.IPv6Any instance to check if the address is detected reliably.
```

Comment:
why not?

Reply:
We special-case the behavior for IPv6Any (see the other change), so this is to make sure we detect the IPv6Any correctly and don't accidentally regress, see JamesNK's comment above

Reply:
The comment in the code should explain because the first question when reading it is "why?"

Reply:
```suggestion
                // Use a copy of IPAddress.IPv6Any to make sure address detection doesn't rely on reference equality comparison.
```
Maybe something like this?